agent/internal/nvidia.go

```diff
@@ -43,17 +45,69 @@ func getNvidiaVersion() (string, error) {
 	return record[0], nil
 }
 
-func nvidiaRuntimeInstalled() (bool, error) {
-	c, err := client.NewClientWithOpts(client.FromEnv)
+// dockerCanUseGPUs determines whether Docker can run containers with GPUs. Unfortunately, it seems
+// that the only way to do so is to actually start a container with a GPU request and see whether it
+// fails with a particular error.
+func dockerCanUseGPUs() (bool, error) {
+	c, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion("1.40"))
 	if err != nil {
-		return false, errors.Wrap(err, "error connecting to docker daemon")
+		return false, errors.Wrap(err, "error connecting to Docker daemon")
 	}
 
-	info, err := c.Info(context.Background())
+	defer func() {
+		if cErr := c.Close(); cErr != nil {
+			log.WithError(cErr).Warn("failed to close Docker connection")
+		}
+	}()
+
+	const testImage = "debian:10-slim"
```

Comment:
I definitely considered that, but figured it behooved me to at least make an effort to avoid regressions in functionality. But yeah, I agree that this is annoying (see "Unfortunately..." above :slightly_smiling_face:). @neilconway, what do you think about just dropping this detection and turning the behavior into "expose GPU slots but fail when we try to start a container using them" when there are GPUs but Docker isn't set up to use them?